subprojects/language-native/src/main/java/org/gradle/language/swift/plugins/SwiftLibraryPlugin.java

```diff
@@ -0,0 +1,122 @@
+import org.gradle.nativeplatform.toolchain.internal.NativeToolChainInternal;
+import org.gradle.nativeplatform.toolchain.internal.NativeToolChainRegistryInternal;
+
+import java.util.Collections;
+
+/**
+ * <p>A plugin that produces a native library from Swift source.</p>
+ *
+ * <p>Assumes the source files are located in `src/main/swift`.</p>
+ *
+ * @since 4.1
+ */
+public class SwiftLibraryPlugin implements Plugin<Project> {
+    @Override
+    public void apply(Project project) {
+        project.getPluginManager().apply(SwiftBasePlugin.class);
+
+        // TODO - extract some common code to setup the compile task and conventions
+        // Add a compile task
+        final SwiftCompile compile = project.getTasks().create("compileSwift", SwiftCompile.class);
+
+        compile.includes(project.getConfigurations().getByName(SwiftBasePlugin.SWIFT_IMPORT_PATH));
+        compile.lib(project.getConfigurations().getByName(CppBasePlugin.NATIVE_LINK));
+
+        ConfigurableFileTree sourceTree = project.fileTree("src/main/swift");
+        sourceTree.include("**/*.swift");
+        compile.source(sourceTree);
+
+        // TODO - should use PIC
+        compile.setCompilerArgs(Lists.newArrayList("-emit-library"));
+        compile.setMacros(Collections.<String, String>emptyMap());
+        compile.setModuleName(project.getName());
+
+        // TODO - should reflect changes to build directory
+        compile.setObjectFileDir(project.file("build/main/objs"));
+
+        DefaultNativePlatform currentPlatform = new DefaultNativePlatform("current");
+        compile.setTargetPlatform(currentPlatform);
+
+        // TODO - make this lazy
+        NativeToolChain toolChain = ((ProjectInternal) project).getModelRegistry().realize("toolChains", NativeToolChainRegistryInternal.class).getForPlatform(currentPlatform);
+        compile.setToolChain(toolChain);
+
+        String libName = ((NativeToolChainInternal) toolChain).select(currentPlatform).getSharedLibraryName("build/lib/" + project.getName());
+        compile.setOutputFile(project.file(libName));
+
+        project.getTasks().getByName(LifecycleBasePlugin.ASSEMBLE_TASK_NAME).dependsOn(compile);
+
+        // TODO - add lifecycle tasks
+
+
+        Configuration apiElements = project.getConfigurations().create("swiftApiElements");
+        apiElements.setCanBeResolved(false);
+        apiElements.getAttributes().attribute(Usage.USAGE_ATTRIBUTE, project.getObjects().named(Usage.class, Usage.SWIFT_API));
+        // TODO - should be lazy and reflect changes to output file
+        apiElements.getOutgoing().artifact(project.file("build/lib"), new Action<ConfigurablePublishArtifact>() {
+            @Override
+            public void execute(ConfigurablePublishArtifact artifact) {
+                artifact.builtBy(compile);
+            }
+        });
+
+        Configuration implementation = project.getConfigurations().getByName(CppBasePlugin.IMPLEMENTATION);
```

Comment:
There's also a lot of duplication here with the C++ library plugin. We can fix this later. Can you add a TODO here perhaps?